cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -848,29 +880,51 @@ impl ManagedEnvironment {
         dot_flox_path: impl AsRef<Path>,
         generation: Option<GenerationId>,
     ) -> Result<Self, EnvironmentError> {
-        let floxmeta = match FloxMeta::open(flox, &pointer) {
-            Ok(floxmeta) => floxmeta,
-            Err(FloxMetaError::NotFound(_)) => {
-                debug!("cloning floxmeta for {}", pointer.owner);
-                FloxMeta::clone(flox, &pointer).map_err(ManagedEnvironmentError::OpenFloxmeta)?
-            },
-            Err(FloxMetaError::CloneBranch(GitRemoteCommandError::AccessDenied))
-            | Err(FloxMetaError::FetchBranch(GitRemoteCommandError::AccessDenied)) => {
+        let _lock = Self::require_floxmeta_lock(&floxmeta_dir(flox, &pointer.owner));
+
+        let existing_floxmeta = match FloxMeta::open(flox, &pointer) {
+            Ok(floxmeta) => Some(floxmeta),
+            Err(FloxMetaError::NotFound(_)) => None,
+            Err(FloxMetaError::FetchBranch(GitRemoteCommandError::AccessDenied)) => {
                 return Err(EnvironmentError::ManagedEnvironment(
                     ManagedEnvironmentError::AccessDenied,
-                ));
+                ))?;
             },
-            Err(FloxMetaError::CloneBranch(GitRemoteCommandError::RefNotFound(_)))
-            | Err(FloxMetaError::FetchBranch(GitRemoteCommandError::RefNotFound(_))) => {
+            Err(FloxMetaError::FetchBranch(GitRemoteCommandError::RefNotFound(_))) => {
                 return Err(EnvironmentError::ManagedEnvironment(
                     ManagedEnvironmentError::UpstreamNotFound {
                         env_ref: pointer.into(),
                         upstream: flox.floxhub.base_url().to_string(),
                         user: flox.floxhub_token.as_ref().map(|t| t.handle().to_string()),
                     },
-                ));
+                ))?;
+            },
+            Err(e) => return Err(ManagedEnvironmentError::OpenFloxmeta(e))?,
+        };
+
+        let floxmeta = match existing_floxmeta {
+            Some(floxmeta) => floxmeta,
+            None => {
+                debug!("cloning floxmeta for {}", &pointer.owner);
+                match FloxMeta::clone(flox, &pointer) {
+                    Ok(floxmeta) => floxmeta,
+                    Err(FloxMetaError::FetchBranch(GitRemoteCommandError::AccessDenied)) => {
+                        return Err(EnvironmentError::ManagedEnvironment(
+                            ManagedEnvironmentError::AccessDenied,
+                        ))?;
+                    },
+                    Err(FloxMetaError::FetchBranch(GitRemoteCommandError::RefNotFound(_))) => {
```

Comment:
Should these be `FloxMetaError::CloneBranch`?

Can you describe why the `FloxMeta::clone` was refactored out to a separate `match`?